.github/workflows/ci.yaml

```diff
@@ -83,10 +84,31 @@ jobs:
       - name: Doc Tests
         run: just doc-tests
 
+  build-python:
+    name: Build Python
+    runs-on: ubuntu-latest-8-cores
+    needs: ["build"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: extractions/setup-just@v1
+      - uses: actions/cache@v3
+        name: Cache
+        with:
+          path: |
+            ~/.cargo/
+            target/
+            .venv/
```

Comment:
I think we want to have two caches: one with the rust dependencies and one with the python virtualenv? I'm a little worried that we'll end up with a messy cache situation...

Reply:
Yeah, I need to figure this out. It's quite messy, agreed. Also, this doesn't work so have to fix it here.